src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -23,6 +23,7 @@
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.FlowControl;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
 using Microsoft.Extensions.Logging;
+using IHttpHeadersHandler = System.Net.Http.IHttpHeadersHandler;
```

Comment:
Let's use different names to avoid this ambiguity.

Reply:
Ok. We'll need to change this name in runtime as well.

Do you have a name suggestion?

Reply:
`IHttpStreamHeadersHandler`?